File: eAutoShop/eAutoShop.Services/Database/Appointment.cs
using System;
using System.Collections.Generic;

namespace eAutoShop.Services.Database;

public partial class Appointment
{
    public int Id { get; set; }

    public int? EmployeeId { get; set; }

    public int CustomerId { get; set; }

    public DateTime ReservationCreatedDate { get; set; }

    public DateTime ReservationDate { get; set; }

    public DateTime? EstimatedCompletionDate { get; set; }

    public DateTime? CompletionDate { get; set; }

    public double TotalAmount { get; set; }

    public int CarModelId { get; set; }

    public TimeOnly TotalDuration { get; set; }

    public string State { get; set; } = null!;

    public string Type { get; set; } = null!;

    public bool DeletedByShop { get; set; }

    public bool DeletedByCustomer { get; set; }

    public string? RejectionReason { get; set; }

    public string? CancellationReason { get; set; }

    public string? ConfirmedBy { get; set; }

    public DateTime? ConfirmedAt { get; set; }

    public string? RejectedBy { get; set; }

    public DateTime? RejectedAt { get; set; }

    public string? CancelledBy { get; set; }

    public DateTime? CancelledAt { get; set; }

    public string? StartedBy { get; set; }

    public DateTime? StartedAt { get; set; }

    public string? CompletedBy { get; set; }

    public virtual ICollection<AppointmentDetail> AppointmentDetails { get; set; } = new List<AppointmentDetail>();

    public virtual CarModel CarModel { get; set; } = null!;

    public virtual User Customer { get; set; } = null!;

    public virtual User? Employee { get; set; }

    public virtual StaffReview? StaffReview { get; set; }
}

File: eAutoShop/eAutoShop.UI/eautoshop_mobile/lib/models/appointment/appointment.dart
import 'package:json_annotation/json_annotation.dart';

part 'appointment.g.dart';

@JsonSerializable()
class Appointment {
  int id;
  int customerId;
  String customerUsername;
  int? employeeId;
  String? employeeUsername;

  @JsonKey(defaultValue: false)
  bool hasStaffReview;

  String carModel;
  String? rejectionReason;
  String? cancellationReason;
  String? confirmedBy;
  DateTime? confirmedAt;
  String? rejectedBy;
  DateTime? rejectedAt;
  String? cancelledBy;
  DateTime? cancelledAt;
  String? startedBy;
  DateTime? startedAt;
  String? completedBy;
  DateTime reservationCreatedDate;
  DateTime reservationDate;
  DateTime? estimatedCompletionDate;
  DateTime? completionDate;
  double totalAmount;
  String totalDuration;
  String state;
  String type;

  Appointment({
    required this.id,
    required this.customerId,
    required this.customerUsername,
    this.employeeId,
    this.employeeUsername,
    this.hasStaffReview = false,
    required this.carModel,
    this.rejectionReason,
    this.cancellationReason,
    this.confirmedBy,
    this.confirmedAt,
    this.rejectedBy,
    this.rejectedAt,
    this.cancelledBy,
    this.cancelledAt,
    this.startedBy,
    this.startedAt,
    this.completedBy,
    required this.reservationCreatedDate,
    required this.reservationDate,
    this.estimatedCompletionDate,
    this.completionDate,
    required this.totalAmount,
    required this.totalDuration,
    required this.state,
    required this.type,
  });

  factory Appointment.fromJson(Map<String, dynamic> json) =>
      _$AppointmentFromJson(json);

  Map<String, dynamic> toJson() => _$AppointmentToJson(this);
}
